### PR TITLE
UI for restoring checkpoints. Fixes #51

### LINKED ISSUE #51: Create a UI to restore neo-express checkpoints

### PR DESCRIPTION
**Pull request by djnicholson**

### CODE REVIEW

**Comment by devhawk** [comment]
checkpoints are specific to a single neo-express instance. in the checkpoint (which is just a zip file) is a file named ADDRESS.neo-express containing the magic number + the script hash of the single consensus node's default wallet account. When you discover a checkpoint, can you retrieve this information so that the user only gets the option to restore checkpoints that are specific to the selected neo-express instance? Maybe also only show the "restore checkpoints" command if there are checkpoints in the workspace for this specific neo-express?

**Comment by djnicholson** [comment]
> checkpoints are specific to a single neo-express instance. in the checkpoint (which is just a zip file) is a file named ADDRESS.neo-express containing the magic number + the script hash of the single consensus node's default wallet account. When you discover a checkpoint, can you retrieve this information so that the user only gets the option to restore checkpoints that are specific to the selected neo-express instance? 

Done

> Maybe also only show the "restore checkpoints" command if there are checkpoints in the workspace for this specific neo-express?

This is possible but I don't think it's worth the complexity. Every time we create a new possibility for options that should or should not display in that right-click menu, I need to add a new contextValue string; this leads to an explosion in package.json where each command has to be listed lots of times for each possible contextValue.  Also I would have to carefully manage the order of actions when a refresh happens (make sure that initially checkpoint detection finishes refreshing before the tree-view starts; always force a tree-view refresh when new checkpoints are found or old ones deleted).  

I do have a friendly message that explains the situation if you click restore for an instance that has no checkpoints.

File: src/extension.ts [hunk 1]
```diff
@@ -258,6 +260,37 @@ export function activate(context: vscode.ExtensionContext) {
...
+                        checkpoint.label);
+                    if (result) {
+                        if (result.isError) {
+                            await vscode.window.showErrorMessage(
+                                'There was an unexpected error restoring the checkpoint:\r\n\r\n' + result.output,
+                                { modal: true });
+                        } else {
+                            await vscode.window.showInformationMessage(
+                                'Checkpoint restored',
+                                { modal: true });
```
Comment (devhawk) [inline src/extension.ts]: should not be modal

**Comment by djnicholson** [comment]
> minor nit about modality of success dialog, otherwise good to go

Will renovate it along with all the other instances in a subsequent PR. Tracking with #107.

**Pull request merged**